server/helpers/youtube-dl/youtube-dl-wrapper.ts

```diff
@@ -145,6 +194,57 @@ class YoutubeDLWrapper {
 
     throw new Error(`Cannot guess path of ${tmpPath}. Directory content: ${directoryContent.join(', ')}`)
   }
+
+  private filterUnavailable (list: YoutubeDLPartialEntry[]) {
+    return list.filter((info) => {
+      switch (info.liveStatus) {
+        case VideoImportLiveStatus.LIVE:
+          logger.warning('%s is currently livestreaming, filtering from import list', info.webpageUrl, lTags())
+          return false
+        case VideoImportLiveStatus.POST_PROCESSING:
+          logger.warning('%s is currently post processing, filtering from import list', info.webpageUrl, lTags())
+          return false
+        case VideoImportLiveStatus.TO_BE_PUBLISHED:
+          logger.warning('%s has not been published yet, filtering from import list', info.webpageUrl, lTags())
+          return false
+      }
+
+      return true
+    })
+  }
+
+  private checkFormats (info: YoutubeDLInfo) {
+    if (info.formats.length === 0) {
+      logger.warning('%s has no downloadable video formats available, skipping', info.webpageUrl, lTags())
+      throw new YoutubeDLError({
+        message: `${info.webpageUrl} has no downloadable video formats available, skipping`,
+        code: YoutubeDLError.CODE.NO_FORMATS_AVAILABLE
+      })
+    }
+  }
+
+  private checkAvailability (info: YoutubeDLInfo | YoutubeDLPartialEntry) {
```

Comment:
Can't you share some code with `filterUnavailable`?

Reply:
Good point. 👍🏻  Will do.

Reply:
DRYified it.